src/daemon/auth/providers/openid_connect/provider.rs

```diff
@@ -338,10 +341,24 @@
                                         }
                                     }
                                     Err(err) => {
-                                        warn!(
-                                            "OpenID Connect: Could not extend login session for user '{}': {}",
-                                            &session.id, err
-                                        );
+                                        match &err {
+                                            // this is where the RFC-6749 5.2 Error Response is received and
+                                            // return to the caller. It's the responsibility of the caller
+                                            // to decide whether to retry or report back to the user.
+                                            openidconnect::RequestTokenError::ServerResponse(r) => {
+                                                warn!(
+                                                    "OpenID Connect: Could not extend login session for user '{}': {}",
+                                                    &session.id, "Connection to OpenID Connect provider not yet established"
```

Comment:
This warning message cannot be correct as if we hadn't already established a connection to the provider we would not be handling a provider response.

Reply:
No, sure. Any suggestions for a better one?